test/stdlib/TestData.swift

```diff
@@ -1172,6 +1172,13 @@ class TestData : TestDataSuper {
         let dataFromSliceOfData = Data(sliceOfData)
         expectEqual(sliceOfData, dataFromSliceOfData)
     }
+
+    func test_reversedDataInit() {
+        let data = Data(bytes: [1, 2, 3, 4, 5, 6, 7, 8, 9])
+        let reversedData = Data(data.reversed())
```

Comment:
@airspeedswift mentioned that the previous test code did not replicate the potential ambiguity. This is a verbatim replication now that hits the initializers and the inference of the return type of reversed.